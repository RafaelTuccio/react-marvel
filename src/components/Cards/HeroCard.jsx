import { useEffect, useState } from "react";
import favoritoOff from '../assets/favorito_02.svg';
import favoritoOn from '../assets/favorito_01.svg';
import { useFavorito } from "../../service/useFavorito";




const HeroCard = (props) => {
    let { hero, favoritar, compareFavoritos, count, setCount } = props;
    let [favorito, setFavorito] = useState(false);
    let {favoritos } = useFavorito()
    let tamanho = "/standard_fantastic.";
    

    useEffect(() => {
       
        if(favoritos){
            setFavorito(compareFavoritos(favoritos, hero.id))
            setCount(favoritos.length);
            
        }
        
        
    },[])

   
    
    
    
     return(
        <div className="card-heroi">
            <img src={hero.thumbnail.path + tamanho + hero.thumbnail.extension} alt="foto heroi" />
            <div className="card-heroi-footer">
                <span>{hero.name}</span>
                <div onClick={() => {
                    favoritar(hero);
                    if(!favorito && count < 5){
                        setFavorito(true)
                        
                    }
                    if(favorito){
                        setFavorito(false)
                    }
                }}>
                        {
                            favorito === false ?
                            <img src={favoritoOff} alt="coracao branco" />
                            :<img src={favoritoOn} alt="coracao vermelho" />
                        }
                    </div>
            </div>
        </div>
    );
}

export default HeroCard;