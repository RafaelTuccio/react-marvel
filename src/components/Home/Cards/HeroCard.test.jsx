import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HeroCard from './HeroCard';


const hero = {
    id: 1011334,
    name: "3-D Man",
    description: "",
    modified: "2014-04-29T14:18:17-0400",
    thumbnail: {
        path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
        extension: "jpg"
    }

}


const index = 1
const count = 1
const setCount = ""


describe("<HeroCard />", () => {
    it("deve renderizar normalmente", () => {
        const fn = jest.fn()
        const fnn = jest.fn()
        const { debug } = render(<HeroCard key={hero.id} 
            hero={hero} index={index} 
            count={count} setCount={setCount} 
            compareFavoritos={fn} 
            favoritar={fnn}/>)

            debug();
    })
    it("deve comparar com o snapshot", () => {
        const fn = jest.fn()
        const fnn = jest.fn()
        const { container } = render(<HeroCard key={hero.id} 
            hero={hero} index={index} 
            count={count} setCount={setCount} 
            compareFavoritos={fn} 
            favoritar={fnn}/>);
        expect(container.firstChild).toMatchSnapshot();
    })

    it("deve funcionar o oneClick", () => {
        const fn = jest.fn()
        render(<HeroCard onClick={fn}/>)
        
        fireEvent.click(<HeroCard />);
        expect(fn).toHaveBeenCalled()
    })
})