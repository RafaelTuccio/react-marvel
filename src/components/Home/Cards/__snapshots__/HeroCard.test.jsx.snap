// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroCard /> deve comparar com o snapshot 1`] = `
<div
  class="card-heroi"
>
  <img
    alt="foto heroi"
    src="http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784/standard_fantastic.jpg"
  />
  <div
    class="card-heroi-footer"
  >
    <span>
      3-D Man
    </span>
    <div>
      <img
        alt="coracao branco"
        src="favorito_02.svg"
      />
    </div>
  </div>
</div>
`;
